errors/handler_test.go

```diff
@@ -2,26 +2,63 @@ package errors_test
 
 import (
 	"errors"
-	errs "github.com/ONSdigital/dp-observation-importer/errors"
+	"testing"
+
+	eventSchema "github.com/ONSdigital/dp-import-reporter/schema"
+
+	errorhandler "github.com/ONSdigital/dp-observation-importer/errors"
+	"github.com/ONSdigital/dp-observation-importer/mocks"
 	"github.com/ONSdigital/go-ns/kafka/kafkatest"
 	. "github.com/smartystreets/goconvey/convey"
-	"testing"
 )
 
+type file struct {
+	instance string
+}
+
 func TestSpec(t *testing.T) {
 
 	Convey("Given an error handler", t, func() {
 
-		outputChannel := make(chan []byte, 1)
-		mockMessageProducer := kafkatest.NewMessageProducer(outputChannel, nil, nil)
-
-		errorHandler := errs.NewKafkaHandler(mockMessageProducer)
-
-		Convey("When handle is called with a nil log data", func() {
+		Convey("When error handler is called ", func() {
+			errHandle := &mocks.HandlerMock{
+				HandleFunc: func(instanceId string, err error) {
+					//
+				},
+			}
+			errHandle.Handle("a4695fee-f0a2-49c4-b136-e3ca8dd40476", errors.New("error"))
+			Convey("And a complete run through should have 1 call to the handle", func() {
+				So(len(errHandle.HandleCalls()), ShouldEqual, 1)
+			})
+		})
+	})
+}
 
-			errorHandler.Handle("123", errors.New("broken"), nil)
+func TestKafkaProducer(t *testing.T) {
+	Convey("Given a new error kafka producer is created", t, func() {
+		Convey("When a new kafka producer is created", func() {
+			outputChannel := make(chan []byte, 1)
+			mockMessageProducer := kafkatest.NewMessageProducer(outputChannel, nil, nil)
 
-			Convey("The error message is written (no panics occur)", func() {
+			errorHandler := errorhandler.NewKafkaHandler(mockMessageProducer)
+			Convey("And the error kafka is not nil", func() {
+				So(errorHandler, ShouldNotBeNil)
+			})
+			errorHandler.Handle("instanceId", errors.New("1"))
```

Comment:
This test feels incomplete - should there be an assertion here? What are we expecting to happen when we call errorHandler.Handle and can we check that it's behaving correctly? Otherwise, should we remove this function call entirely?